qiime2/sdk/usage.py

```diff
@@ -284,30 +691,50 @@ def action(self, action: UsageAction, inputs: UsageInputs,
         computed_outputs = self._action_(action, input_opts, output_opts)
         self._add_outputs_to_scope(outputs, computed_outputs)
 
-    def _action_(self, action: UsageAction,
-                 input_opts: dict, output_opts: dict):
+    def _action_(
+        self, action: UsageAction, input_opts: dict, output_opts: dict
+    ) -> dict:
         raise NotImplementedError
 
     def _assert_has_line_matching_(self, ref, label, path, expression):
         raise NotImplementedError
 
-    def get_result(self, ref):
+    def get_result(self, ref: str) -> ScopeRecord:
+        """
+        Get the record for a Usage example output. This is a convenience
+        method used to access records generated after running `Usage.action`.
+
+        Parameters
+        ----------
+        ref : str
+            Output name
+
+        Raises
+        ------
+            If `ref` is not associated with a record generated by
```

Comment:
Missing the exception type (probably a `KeyError`)

Reply:
Good catch!  We went with `TypeError` in #536, here and in other places as well.  You good with sticking with that?

Reply:
This method raises both a `KeyError` (by way of `_get_record`) and a `TypeError`, depending on nature of the failure, you will need to account for both in the docs.

Reply:
This is still missing the `KeyError` I mentioned in this thread. Please see https://github.com/qiime2/qiime2/blob/3953ce918683e7af7012a8d8570ca73f3a486970/qiime2/sdk/usage.py#L214 for more details.

Reply:
I think I just mistakenly called it a `ValueError`. 

>Why do you think the TypeError should be kept over the KeyError?

The `KeyError` seems more obvious to me.  My (unjustified) interpretation of "errors that are non-obvious or have a large chance of getting raised," was that, if it's obvious, leave it out.